eiffel-syntax-and-usage/compositions-and-validity-checking.md

```diff
@@ -51,4 +51,12 @@ _Isn't this a reinvention of the wheel?_ you may ask. After all, there are plent
 
 It should also be noted that Eiffel's dependency syntax is opinionated. An interface requiring supporting implementation does not dictate that it requires any particular implementation (actually it can - simply use __data.dependsOn__). Rather, it is up to that implementation to declare that it supports the interface. This is in order to encourage separation of concerns and decoupling.
 
-Furthermore, Eiffel's dependency syntax makes no distinction between e.g. build time and runtime dependencies. This is for two reasons. First, it is not intended for low level dependency management: most programming languages have dedicated and specialized tools that will handle this type of dependency management. Instead, where Eiffel's comes into play is in tracking and validating large, sprawling systems of heterogeneous artifacts. Second, even if one attempts to develop specific syntax each type of dependency there will always be corner use cases which are not covered: better then to err on the side of abstraction, rather than implicitly ruling out use cases by being overly specific. That being said, it is also generally the case that an artifact has a fair idea of its intended use - it is rare that one and the same artifact is both a deployable service and a source code library, with the need to declare a unique set of dependencies for each case. Hence, the EiffelArtifactCreatedEvent dependency declarations shall be read in context.
+Furthermore, Eiffel's dependency syntax makes no distinction between e.g. build time and runtime dependencies. This is
+for two reasons. First, it is not intended for low level dependency management: most programming languages have
+dedicated and specialized tools that will handle this type of dependency management. Instead, where Eiffel's comes into
+play is in tracking and validating large, sprawling systems of heterogeneous artifacts. Second, even if one attempts to
+develop specific syntax each type of dependency there will always be corner use cases which are not covered: better then
+to err on the side of abstraction, rather than implicitly ruling out use cases by being overly specific. That being
+said, it is also generally the case that an artifact has a fair idea of its intended use - it is rare that one and the
+same artifact is both a deployable service and a source code library, with the need to declare a unique set of
+dependencies for each case. Hence, the EiffelArtifactCreatedEvent dependency declarations shall be read in context.
```

Comment:
It's a bit weird that we're mixing very long unwrapped lines with nicely wrapped paragraphs like this one, but I suppose wrapping everything would result in total mayhem. I personally prefer wrapping so I think this is a step in the right direction.

Reply:
True! I wanted to do something for now so I set the line length to 900. When trying to set to 120 it would affect 500 lines so that is probably best done in a PR of its own.